transport/tchannel/header.go

```diff
@@ -98,15 +98,22 @@ func writeRequestHeaders(
 	format tchannel.Format,
 	appHeaders map[string]string,
 	getWriter func() (tchannel.ArgWriter, error),
+	rawHeader bool,
 ) error {
 	headers := transport.NewHeadersWithCapacity(len(appHeaders))
 	// TODO: zero-alloc version
-
 	for k, v := range appHeaders {
 		headers = headers.With(k, v)
 	}
 
-	return writeHeaders(format, headers, getWriter)
+	var items map[string]string
+	if rawHeader {
+		items = headers.ForwardingItems()
+	} else {
+		items = headers.Items()
+	}
```

Comment:
You can use the original `appHeaders` if `rawHeaders` is true, otherwise create a copy of the map with the canonicalize function used to map the keys.